File: dataset/human/python/file_3336.py
n = int(input())
def prime_decomposition(n):
  i = 2
  table = []
  while i * i <= n:
    while n % i == 0:
      n /= i
      table.append(i)
    i += 1
  if n > 1:
    table.append(n)
  return table


a = prime_decomposition(n)


d ={}
for i in a:
    if i not in d:
        d[i] = int(0)
    d[i] += 1
    


c = list(d.values())


l =[]
for i in c:
    e = 1
    count = 1
    f = 0
    while i >=  e:
        count += 1
        f += 1 
        e = e + count
    l.append(f)
    
print(sum(l))

File: dataset/human/cpp/file_3336.cpp
#include<iostream>
#include<string>
#include<vector>
#include<algorithm>
#include<utility>
#include<algorithm> // sort
#include<map> // pair
#include<cmath>
using namespace std;
using ll = long long;
const ll mod = 1e9+7;
int main(){
  ll n,x,count=0,sum=0;
  cin >> n;
  ll a[n];
  bool havezero = false;
  for(int i=0;i<n;i++){
    cin >> x;
    a[i] = abs(x);
    if(x < 0){
      count++;
    }else if(x == 0){
      havezero = true;
    }
    sum += abs(x);
  }
  //cout << count << endl;
  if(havezero || count % 2 == 0){
    cout << sum;
  }else{
    sort(a,a+n);
    //cout << a[0] << endl;
    cout << sum - a[0] * 2;
  }
}
